astropy/utils/console.py

```diff
@@ -274,10 +276,11 @@ def _write_with_fallback(s, write, fileobj):
     """
     try:
         write(s)
-        return write
     except UnicodeEncodeError:
         # Let's try the next approach...
         pass
+    else:
```

Comment:
Again, here the `else` clause hides the purpose of the `except one` - not an improvement.